cmd/antrea-agent/agent.go

```diff
@@ -384,29 +387,35 @@ func run(o *Options) error {
 		proxyAll := o.config.AntreaProxy.ProxyAll
 		skipServices := o.config.AntreaProxy.SkipServices
 		proxyLoadBalancerIPs := *o.config.AntreaProxy.ProxyLoadBalancerIPs
+		optsModifier, err := getServiceListOptionsModifier(o.config.AntreaProxy.ServiceProxyName)
```

Comment:
Could we move the code in this block to the proxy package? It became long and includes quite some proxy specific logic.

Including: getServiceListOptionsModifier, NewProxier() calls, proxyInformerFactory creation, etc.

Reply:
I have pushed a new commit, could you help check that is that your expectation? @jianjuns Thanks.